sharding-proxy/sharding-proxy-frontend/sharding-proxy-frontend-mysql/src/main/java/org/apache/shardingsphere/shardingproxy/frontend/mysql/auth/MySQLAuthenticationEngine.java

```diff
@@ -60,9 +63,10 @@ public boolean auth(final ChannelHandlerContext context, final PacketPayload pay
             backendConnection.setUserName(response41.getUsername());
             context.writeAndFlush(new MySQLOKPacket(response41.getSequenceId() + 1));
         } else {
-            // TODO localhost should replace to real ip address
+            SocketAddress socketAddress = context.channel().remoteAddress();
+            String hostAddress = socketAddress instanceof InetSocketAddress ? ((InetSocketAddress) socketAddress).getAddress().getHostAddress() : socketAddress.toString();
```

Comment:
What if the socketAddress is null? You can find the null value possibility in the comment of API.

Reply:
I find “null if this channel is not connected” in the API comment, but this method is called by auth method, i think the channel is connected

Reply:
Yes, you are right.